components/build_image.py

```diff
@@ -0,0 +1,118 @@
+import argparse
+import datetime
+import logging
+import os
+import subprocess
+import sys
+import time
+import yaml
+
+
+def run(command, cwd=None, env=None, polling_interval=datetime.timedelta(seconds=1)):
+  """Run a subprocess.
+  Copied from kubeflow/test so it's easier to run locally.
```

Comment:
Hopefully in the future we will be able to organize our codebase better so that we don't have to duplicate code like this.